pkg/cmd/infra/deployer/deployer.go

```diff
@@ -78,15 +80,29 @@ func NewCommandDeployer(name string) *cobra.Command {
 
 // deploy executes a deployment strategy.
 func deploy(kClient kclient.Interface, namespace, deploymentName string) error {
-	newDeployment, oldDeployments, err := getDeployerContext(&realReplicationControllerGetter{kClient}, namespace, deploymentName)
-
+	deployment, oldDeployments, err := getDeployerContext(&realReplicationControllerGetter{kClient}, namespace, deploymentName)
```

Comment:
Why does this method retrieve the old replication controllers and then throw them away so that another client has to go load them?